### PR TITLE
Add collector for ssacli

### PR DESCRIPTION
**Pull request by sudeephb**

### CODE REVIEW

File: prometheus_hardware_exporter/collector.py [hunk 1]
```diff
@@ -733,3 +734,113 @@ def fetch(self) -> List[Payload]:
...
+            Specification(
+                name="ssacli_controllers",
+                documentation="Total number of controllers",
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_controller_status",
+                documentation="Indicates the status of controller part",
+                labels=["slot", "part"],
+                metric_class=GaugeMetricFamily,
```
Comment (Pjack) [inline prometheus_hardware_exporter/collector.py:761]: Please use InfoMetricFamily and the naming could be ssacli_controller_info
Sample code: https://github.com/canonical/prometheus-hardware-exporter/blob/main/prometheus_hardware_exporter/collector.py#LL52C30-L52C46

File: prometheus_hardware_exporter/collector.py [hunk 2]
```diff
@@ -733,3 +734,113 @@ def fetch(self) -> List[Payload]:
...
+                name="ssacli_physical_drives",
+                documentation="The number of physical drives in the slot",
+                labels=["slot"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_ld_status",
+                documentation="Indicates the status of logical drive",
+                labels=["slot", "drive_id"],
+                metric_class=GaugeMetricFamily,
```
Comment (Pjack) [inline prometheus_hardware_exporter/collector.py:779]: Please use InfoMetricFamily and the naming could be ssacli_ld_info

File: prometheus_hardware_exporter/collector.py [hunk 3]
```diff
@@ -733,3 +734,113 @@ def fetch(self) -> List[Payload]:
...
+            ),
+            Specification(
+                name="ssacli_pd_status",
+                documentation="Indicates the status of physical drive",
+                labels=["slot", "drive_id"],
+                metric_class=GaugeMetricFamily,
```
Comment (Pjack) [inline prometheus_hardware_exporter/collector.py:785]: Please use InfoMetricFamily and the naming could be ssacli_pd_info

File: prometheus_hardware_exporter/collector.py [hunk 4]
```diff
@@ -733,3 +734,113 @@ def fetch(self) -> List[Payload]:
...
+        ]
+
+        for slot, payload in ssacli_payload.items():
+            ctrl_status = payload["controller_status"]
+            for part, status in ctrl_status.items():
+                payloads.append(
+                    Payload(
+                        name="ssacli_controller_status",
+                        labels=[slot, part],
+                        value=self._get_status_value(status),
```
Comment (Pjack) [inline prometheus_hardware_exporter/collector.py:808]: The design seems different than other collectors. 
https://github.com/canonical/prometheus-hardware-exporter/blob/main/prometheus_hardware_exporter/collector.py#L143

Do we want to align the design ?

Solution 1: state as an info in label
Solution 2: status as a value , 1: healthy, 0: unhealthy

**Review by Pjack — CHANGES_REQUESTED** [review]

**Review by Pjack — APPROVED** [review]
LGTM

File: prometheus_hardware_exporter/collector.py [hunk 5]
```diff
@@ -733,3 +734,101 @@ def fetch(self) -> List[Payload]:
...
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_physical_drives",
+                documentation="The number of physical drives in the slot",
+                labels=["slot"],
+                metric_class=GaugeMetricFamily,
+            ),
+            Specification(
+                name="ssacli_ld",
```
[the remark this answers is not in the document]
Comment (Pjack) [reply prometheus_hardware_exporter/collector.py:775]: +1

**Review by jneo8 — APPROVED** [review]

**Pull request merged**